api/src/main/java/org/itsallcode/openfasttrace/api/core/LinkedSpecificationItem.java

```diff
@@ -281,23 +324,44 @@ public boolean isCoveredShallow()
         return this.getCoveredArtifactTypes().containsAll(this.getNeedsArtifactTypes());
     }
 
+    /**
+     * @return <code>true</code> if all needed attribute types are covered by approved items
+     */
+    public boolean isCoveredShallowWithApprovedItems()
+    {
+        return getStatus() == ItemStatus.APPROVED && this.getCoveredApprovedAttributeTypes()
```

Comment:
Extract method `isApproved()` from the first part of the condition.
Extract method `areAllCoveredArtifactTypesApproved()`.

Reply:
DONE